### ISSUE TITLE
docker exec -it with no TTY leaks execids

### ISSUE THREAD

**Issue by cyphar** [moby/moby#33084]
**NOTE** I've already debugged and fixed this problem. I'll submit a patch later (though it'll land in https://github.com/docker/cli).

**Steps to reproduce the issue:**
1. Do `docker exec -it` in a situation with no TTY.
2. Inspect the container you're execing.

**Describe the results you received:**
The number of execIDs grows.

**Describe the results you expected:**
There should be none.

**Additional information you deem important (e.g. issue happens only occasionally):**

**Output of `docker version`:**

```
Client:
 Version:      17.04.0-ce
 API version:  1.28
 Go version:   go1.7.5
 Git commit:   78d1802
 Built:        Thu Apr 27 14:10:53 2017
 OS/Arch:      linux/amd64

Server:
 Version:      17.04.0-ce
 API version:  1.28 (minimum version 1.12)
 Go version:   go1.7.5
 Git commit:   78d1802
 Built:        Thu Apr 27 14:10:53 2017
 OS/Arch:      linux/amd64
 Experimental: false
```

**Output of `docker info`:**

```
Containers: 12
 Running: 0
 Paused: 0
 Stopped: 12
Images: 186
Server Version: 17.04.0-ce
Storage Driver: devicemapper
 Pool Name: docker-0:41-8367155-pool
 Pool Blocksize: 65.54kB
 Base Device Size: 10.74GB
 Backing Filesystem: ext4
 Data file: /dev/loop0
 Metadata file: /dev/loop1
 Data Space Used: 14.59GB
 Data Space Total: 107.4GB
 Data Space Available: 88.35GB
 Metadata Space Used: 14.86MB
 Metadata Space Total: 2.147GB
 Metadata Space Available: 2.133GB
 Thin Pool Minimum Free Space: 10.74GB
 Udev Sync Supported: true
 Deferred Removal Enabled: false
 Deferred Deletion Enabled: false
 Deferred Deleted Device Count: 0
 Data loop file: /var/lib/docker/devicemapper/devicemapper/data
 Metadata loop file: /var/lib/docker/devicemapper/devicemapper/metadata
 Library Version: 1.03.01 (2017-04-13)
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins: 
 Volume: local
 Network: bridge host macvlan null overlay
Swarm: inactive
Runtimes: oci runc
Default Runtime: runc
Init Binary: 
containerd version:  (expected: 422e31ce907fd9c3833a38d7b8fdd023e5a76e73)
runc version: N/A (expected: 9c2d8d184e5da67c95d601382adf14862e4f2228)
init version: N/A (expected: 949e6facb77383876aeff8a6944dde66b3089574)
Security Options:
 apparmor
 seccomp
  Profile: default
Kernel Version: 4.10.12-1-default
Operating System: openSUSE Tumbleweed
OSType: linux
Architecture: x86_64
CPUs: 4
Total Memory: 7.69GiB
Name: baradur
ID: DOHI:RAQ5:MYV5:73JL:7UHU:6NXS:RWGD:IQ7D:IQZ2:COCW:BWJ6:HHYN
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): false
Registry: https://index.docker.io/v1/
Experimental: false
Insecure Registries:
 127.0.0.0/8
Live Restore Enabled: false
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**

**Comment by thaJeztah** [comment]
Thanks @cyphar

**Comment by cyphar** [comment]
My patch to fix this is in docker/cli#52.
